tools/data-handler/src/utils/clingo-fact-builder.ts

```diff
@@ -0,0 +1,134 @@
+export type AllowedClingoType = string | number | boolean;
+
+export type NestedBuilder = (builder: ClingoFactBuilder) => ClingoFactBuilder;
+
+export type ClingoArgument =
+  | AllowedClingoType
+  | AllowedClingoType[]
+  | NestedBuilder
+  | ClingoFactBuilder;
+
+type ClingoArgumentInternal =
+  | AllowedClingoType
+  | AllowedClingoType[]
+  | ClingoFactBuilder;
+
+/**
+ * This function takes care of encoding chars, which might produce issues in clingo
+ * This should be done for user provided values
+ */
+export function encodeClingoValue(value: string) {
+  return value.replace(/[\n\\"]/g, (char) => {
+    if (char === '\n') {
+      return '\\n';
+    }
+    return `\\${char}`;
+  });
+}
+
+export class ClingoFactBuilder {
+  protected predicate: string;
+  private end: string;
+  private arguments: ClingoArgumentInternal[] = [];
+
+  constructor(predicate: string, end: string = '.') {
+    this.predicate = predicate;
+    this.end = end;
+  }
+
+  /**
+   * Adds an argument, which is any value
+   * @param arg
+   * @returns
+   */
+  addArgument(arg: ClingoArgument | null): ClingoFactBuilder {
+    if (arg === null) {
+      return this;
+    }
+    if (typeof arg === 'function') {
+      const nestedBuilder = new ClingoFactBuilder('', '');
+      this.arguments.push(arg(nestedBuilder));
+    } else if (arg instanceof ClingoFactBuilder) {
```

Comment:
This `else if` is not needed (does the same as `else`)

Reply:
done